sycl/source/detail/pi.cpp

```diff
@@ -292,6 +292,17 @@ const vector_class<plugin> &initialize() {
     // time when "Plugins" could be deleted.
     Plugins = new vector_class<plugin>;
     initializePlugins(Plugins);
+    // TODO: remove the following code when we remove the support for legacy
+    // env vars.
+    // Emit the deprecation warning message if SYCL_BE or SYCL_DEVICE_TYPE is
+    // set.
+    if (SYCLConfig<SYCL_BE>::get() || getenv("SYCL_DEVICE_TYPE")) {
```

Comment:
Wouldn't it be better to move this code to where these templates are specialized, i.e. in source/detail/config.hpp? There is already needed logic there to only print it once.

Reply:
I need to check both SYCL_BE and SYCL_DEVICE_TYPE.
Config.hpp only handles SYCL_BE.
Note that this patch only lives a few releases until we really remove the support for SYCL_BE and SYCL_DEVICE_TYPE.
After removal, I will also remove this patch from pi.cpp.

Reply:
Done. I moved the warning message code into config.hpp.